libraries/streaming-e2e-tests/build/create-react-app.yml

```diff
@@ -1,40 +1,19 @@
-# Create a React app that uses ReactWebChat and a customized DirectLineJS
-# that uses that latest version of botframework-streaming
+# Create a React app that uses ReactWebChat and DLJS with the
+# latest nightly version of botframework-streaming
 
 steps:
-  - task: Cache@2
-    displayName: 'Cache create-react-app + WebChat Dependencies'
-    inputs:
-      key: '"CRA-WebChat4.10.1-Cache" | npm | "$(Agent.OS)"'
-      path: $(ReactProjectDir)/node_modules
-      cacheHitVar: 'CacheRestored'
-
-  - powershell: npm install
+  - script: npm install
     displayName: npm install in React App
-    condition: ne(variables.CacheRestored, 'true')
     workingDirectory: $(ReactProjectDir)
 
-  - powershell: npm install react-scripts@latest
-    displayName: npm install react-scripts@latest
-    workingDirectory: $(ReactProjectDir)
+  - script: 'rm -rf */**/botframework-streaming'
```

Comment:
Removing `node_modules` and rerunning `npm install` means:

1. `npm install` will follow `package-lock.json`, so it may not bump to latest
2. If you don't commit `package-lock.json`, few hours/days/weeks later, running `npm install` will produce a very different `node_modules`
   - I.e. it hurt test reproducibility, i.e. "I swear it was working!"
   - And you have no ideas why it was working and why it isn't working now (and vice versa)

Reply:
We actually don't run `npm install` in the `react-app` package after this. The reason I omitted lock files is so references like `latest` and `next` pick up the proper `botbuilder-*` packages. We already do this in the pipeline, so the presumed integrity of package lock files is already compromised.

I'll restrict the versions of other dependencies (by removing `^` and `~` from the version references) so that builds are more reproducible. The end goal is to link against the packages inside the repo instead of pulling down only slightly stale packages from npm.

Reply:
Without the lock file, not only "references like `latest` and `next` pick up the proper `botbuilder-*` packages"... every deps including `uuid`, `node-fetch`, `mocha`, and **everything**, could pick up a latter version.

`^`, `~` and pinning, will only helps with direct dependencies. For transient dependencies, it can still pick up a latter version.

For example, you do pin `node-fetch@1.0.0`, but `node-fetch` only semver to `uuid@^2.0.0` (out of your control). If you don't have `package-lock.json`, hours later, you will pick up `uuid@2.0.9` and it can break your work. And you have no trace on what version of `uuid` you installed and was working, because it could be `uuid@2.0.2`.

Please read about why `package-lock.json` is highly recommended to commit in source branch.